### PR TITLE
feat(core/web-monetization): support adding monetization meta tag

### PR DESCRIPTION
**Pull request by sidvishnoi**
Part of https://github.com/w3c/respec/issues/2806
Wiki: https://github.com/w3c/respec/wiki/monetization

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Looks good so far...

File: src/core/web-monetization.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * This module adds a "monetization" meta-tag to enable web-monetization.
+ *
+ * The meta-tag is added only to "live" documents, and is removed from generated
+ * static documents.
+ */
+import { html } from "./import-maps.js";
+
+export const name = "core/web-monetization";
+
+const DEFAULT_PAYMENT_POINTER = "$wallet.example.com/respec";
```
Comment (marcoscaceres) [inline src/core/web-monetization.js:12]:
```suggestion
const DEFAULT_PAYMENT_POINTER = "$coil.xrptipbot.com/XzUlVjPFQ7qxCPcQqzhbdQ";
```

File: src/core/web-monetization.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    return;
+  }
+
+  const paymentPointer =
+    typeof monetization === "string" ? monetization : DEFAULT_PAYMENT_POINTER;
+
+  document.head.append(html`<meta
+    name="monetization"
+    content="${paymentPointer}"
+    class="removeOnSave"
```
Comment (marcoscaceres) [inline src/core/web-monetization.js:26]: Cash in everywhere... 

```suggestion
```

File: tests/spec/core/web-monetization-spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const metaTag = doc.querySelector("meta[name='monetization']");
+    expect(metaTag).toBeNull();
+  });
+
+  it("is enabled by default", async () => {
+    const ops = makeStandardOps();
+    const doc = await makeRSDoc(ops);
+
+    const metaTag = doc.querySelector("meta[name='monetization']");
+    expect(metaTag.content).toBe("$wallet.example.com/respec");
```
Comment (sidvishnoi) [inline tests/spec/core/web-monetization-spec.js:19]:
```suggestion
    expect(metaTag.content).toBe("$coil.xrptipbot.com/XzUlVjPFQ7qxCPcQqzhbdQ");
```

File: tests/spec/core/web-monetization-spec.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    expect(metaTag.classList).toContain("removeOnSave");
+  });
+
+  it("uses default payment pointer when set to `true`", async () => {
+    const ops = makeStandardOps({ monetization: true });
+    const doc = await makeRSDoc(ops);
+
+    const metaTag = doc.querySelector("meta[name='monetization']");
+    expect(metaTag.content).toBe("$wallet.example.com/respec");
```
Comment (sidvishnoi) [inline tests/spec/core/web-monetization-spec.js:28]:
```suggestion
    expect(metaTag.content).toBe("$coil.xrptipbot.com/XzUlVjPFQ7qxCPcQqzhbdQ");
```

File: src/core/web-monetization.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * This module adds a "monetization" meta-tag to enable web-monetization.
+ *
+ * The meta-tag is added only to "live" documents, and is removed from generated
+ * static documents.
+ */
+import { html } from "./import-maps.js";
+
+export const name = "core/web-monetization";
+
+const DEFAULT_PAYMENT_POINTER = "$coil.xrptipbot.com/XzUlVjPFQ7qxCPcQqzhbdQ";
```
Comment (marcoscaceres) [inline src/core/web-monetization.js:12]:
```suggestion
const DEFAULT_PAYMENT_POINTER = "$ilp.uphold.com/DwJmxPHHi8K3";
```

File: src/core/web-monetization.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    return;
+  }
+
+  const paymentPointer =
+    typeof monetization === "string" ? monetization : DEFAULT_PAYMENT_POINTER;
+
+  document.head.append(html`<meta
+    name="monetization"
+    content="${paymentPointer}"
+  />`);
```
Comment (marcoscaceres) [inline src/core/web-monetization.js:26]:
```suggestion
    content="${paymentPointer}"
    class="removeOnSave"
  />`);
```

File: src/core/web-monetization.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  const paymentPointer =
+    typeof monetization === "string" ? monetization : DEFAULT_PAYMENT_POINTER;
+
+  document.head.append(html`<meta
+    name="monetization"
+    content="${paymentPointer}"
+    class="removeOnSave"
+  />`);
```
Comment (marcoscaceres) [inline src/core/web-monetization.js:24]: Might need a prettier tidy after this: 

```suggestion
  document.head.append(html`
    <!-- This supports ReSpec's development and helps pay the bills - https://opencollective.com/respec --> 
    <meta
    name="monetization"
    content="${paymentPointer}"
    class="removeOnSave"
  />`);
```

**Review by marcoscaceres — APPROVED** [review]
LGTM

**Pull request merged**